docs/preflight/v1beta3-guide.md

```diff
@@ -0,0 +1,487 @@
+  memory:
+    enabled: true
+    minGi: 8
+    recommendedGi: 16
+  ephemeral:
+    enabled: true
+    minGi: 20
+    recommendedGi: 50
+
+workloads:
+  deployments:
+    enabled: true
+    namespace: "default"
+    name: "example-deploy"
+    minReady: 1
+
+databases:
+  postgres:
+    enabled: true
+    collectorName: "postgres"
+    uri: "postgres://user:pass@postgres:5432/db?sslmode=disable"
+  mysql:
+    enabled: true
+    collectorName: "mysql"
+    uri: "mysql://user:pass@tcp(mysql:3306)/db"
+```
+
+
+### Render, run, and extract docs
+
+You can render templates, run preflights with values, and extract requirement docs without running checks.
+
+- **Render a templated preflight spec** to stdout or a file:
+  ```bash
+  preflight template v1beta3.yaml \
+    --values values-base.yaml \
+    --values values-prod.yaml \
+    --set storage.className=fast-local \
+    -o rendered-preflight.yaml
+  ```
+
+- **Run preflights with values** (values and sets also work with `preflight` root command):
+  ```bash
+  preflight run rendered-preflight.yaml
+  # or run directly against the template with values
+  preflight run v1beta3.yaml --values values-prod.yaml --set cluster.minNodes=5
+  ```
+
+- **Extract only documentation** from enabled analyzers in one or more templates:
+  ```bash
+  preflight docs v1beta3.yaml other-spec.yaml \
+    --values values-prod.yaml \
+    --set kubernetes.enabled=true \
+    -o REQUIREMENTS.md
+  ```
+
+Notes:
+- Multiple `--values` files are merged in order; later files win.
+- `--set` uses Helm-style semantics for nested keys and types, applied after files.
+
+
+### Authoring checklist
+
+- Add `docString` with Title, Requirement bullets, rationale, and links.
+- Gate optional analyzers with `{{- if .Values.analyzers.<feature>.enabled }}`.
+- Parameterize thresholds and names with `.Values` expressions.
+- Ensure all required values are present in your values files since there are no fallback defaults.
+- Use precise, user-actionable `message` text for each outcome; add `uri` where helpful.
+- Prefer a minimal values file with everything disabled, and a full values file enabling most checks.
+- Test with `preflight template` (no values, minimal, full) and verify `preflight docs` output reads well.
+
+
+### Example skeleton to start a new spec
+
+```yaml
+apiVersion: troubleshoot.sh/v1beta3
+kind: Preflight
+metadata:
+  name: your-product-preflight
+spec:
+  {{- /* Determine if we need explicit collectors beyond always-on clusterResources */}}
+  {{- $needExtraCollectors := or .Values.databases.postgres.enabled .Values.http.enabled }}
+
+  collectors:
+    # Always collect cluster resources to support core analyzers
+    - clusterResources: {}
+
+    {{- if .Values.databases.postgres.enabled }}
+    - postgres:
+        collectorName: '{{ .Values.databases.postgres.collectorName }}'
+        uri: '{{ .Values.databases.postgres.uri }}'
+    {{- end }}
+
+  analyzers:
+    {{- if .Values.clusterVersion.enabled }}
```

Comment:
Again, this might just be me nitpicking, but seeing this template reinforces to me:

1. We should make sure we're consistent throughought the docs in the values schema that we're using. Here its `.Values.clusterVersion` and in other places its `.Values.kubernetes.version`
2. We should probably be intentional about structuring values files in our documentation in a way that's more representative to how we'd instruct people to write them. For example i'd advise someone to nest preflight specific values like this under a single key so you don't pollute the top level.